addons/escoria-core/game/core-scripts/esc/esc_object_manager.gd

```diff
@@ -99,6 +99,9 @@ func unregister_object(object: ESCObject) -> void:
 	if not escoria.inventory_manager.inventory_has(object.global_id) \
 			and not object.global_id in RESERVED_OBJECTS:
 		objects.erase(object.global_id)
+	else:
```

Comment:
Probably doesn't really matter, but maybe add the "not object.global_id in RESERVED_OBJECTS" as an if clause here, so that *only* inventory objects get re-instanced.

Reply:
Good point!